inst/shiny-examples/CiteSource/app.R

```diff
@@ -241,490 +156,162 @@ server <- function(input, output, session) {
-        select(-author, -year) %>%
-        select(pub_id, everything())
-      
-      db_comparison <- left_join(db_comparison, pub_id, by="duplicate_id")
-    }
+  plotInput <- reactive({
+    source_comparison <- CiteSource::compare_sources(rv$unique, comp_type = input$comp_type)
+    plot_source_overlap_upset(source_comparison, decreasing = c(TRUE, TRUE))
     
-    
-    comparison <- source_comparison(unique_citations())
-    
-    DT::datatable(comparison,
-                  options = list(paging = FALSE, 
-                                 searching = FALSE),
-                  rownames = FALSE)
   })
   
-  # output results table from deduplication pre/post numbers per source and overall
-  output$dedup_results_table <- function() {
-    
-    
-    req(unique_citations())
-    
-    rem_dup_word <- function(x){
-      x <- tolower(x)
-      paste(unique(trimws(unlist(strsplit(x,split=", ",fixed=F,perl=T)))),collapse = 
-              ", ")
-    }
-    
-    # pre-dedup table
-    pre_dedup <- rv$upload_df %>%
-      group_by(cs_source) %>%
-      count()  %>%
-      pivot_wider(names_from=c(cs_source), values_from=n) 
-    
-    #post-dedup table
-    post_dedup <- unique_citations() %>%
-      select(duplicate_id, label, source, record_id) %>%
-      separate_rows(c(source, label), convert = TRUE, sep=", ") %>%
-      unique() %>%
-      pivot_wider(id_col = duplicate_id, names_from=c(source), values_from=c(record_id),
-                  values_fn = list(record_id = length)) %>%
-      select_if(is.numeric) %>%
-      summarise(across(where(is.numeric), ~ sum(.x, na.rm = TRUE))) 
-    
-    combined_dedup_table <- rbind(pre_dedup, post_dedup)
-    combined_dedup_table <- cbind(combined_dedup_table, 
-                                  Total = rowSums(combined_dedup_table)) 
-    
-    combined_dedup_table$Total[2] <- paste0(nrow(unique_citations()), " merged unique citations")
-    
-    combined_dedup_table %>%
-      knitr::kable("html",  caption = "De-duplication results") %>%
-      kable_styling("striped", full_width = T)  %>%
-      pack_rows(
-        index = c("Before deduplication" = 1, "After deduplication" = 1))
-    
-    
-  }
-  
-  
-  #### end section #### year,
-  
-  
-  
-  
-  #### Visualise section ####
-  #render check box ui based on uploaded sources and tags
-  output$checkbox1 <- renderUI({
-    checkboxGroupInput('source_check', 'Sources', unique(rv$df$source))
-  })
-  output$checkbox2 <- renderUI({
-    checkboxGroupInput('tag_check', 'Tags', unique(rv$df$tag))
-  })
-  
-  
-  #### end section ####
-  
-  
-  #### Export section ####
-  
-  #### end section ####
+  output$plotgraph2<-shiny::renderPlot({
+    print(plotInput())
+    })
+  
+   output$downloadPlot <- shiny::downloadHandler(
+     filename = function() { paste("upset", '.png', sep='') },
+     content = function(file) {
+       png(file)
+       print(plotInput())
+       dev.off()
+        }
+  )
+   
```

Comment:
Do you want to add the `citation_summary_table()` - [example](https://www.eshackathon.org/CiteSource/articles/citesource_example.html#step-4a-review-a-summary-table) - here? Would seem to fit into the visualisation tab (or analysis tab if you deem that a more fitting title) ... and we should prob have that table somewhere.

Reply:
Added issue #79